Stack/Opc.Ua.Core/Stack/Server/EndpointBase.cs

```diff
@@ -848,6 +939,19 @@ private void OnProcessRequest(object state)
                     // set the context.
                     SecureChannelContext.Current = m_context;
 
+                    // check if AdditionalHeader field from the request contains trace information
+                    if (m_request.RequestHeader.AdditionalHeader != null && m_request.RequestHeader.AdditionalHeader.Body is AdditionalParametersType parameters)
+                    {
+                        // extract trace information from the request header
+                        var traceContext = ExtractTraceContextFromParameters(parameters);
```

Comment:
it looks like from the trace context only activitycontext is used? can you optimize the funtion to only work through the required results?